### DISCUSSION TITLE
Bug: when using TestClient for testing an endpoint

### DISCUSSION THREAD

**Discussion by dnutiu** [fastapi/fastapi#11486] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@pytest.mark.asyncio
async def test_something(test_client):
    response = test_client.get("/v1/rules/?custom=example.com&args=0&kwargs=0")
    assert response is not None
```

### Description

I need to pass args and kwargs otherwise the test client returns 422. I've encountered the bug before where the test client returned 422 when x was not passed.

### Operating System

macOS

### FastAPI Version

0.110.2

### Pydantic Version

2.7.0

### Python Version

Python 3.12.1

### Additional Context

b'{"detail":[{"type":"missing","loc":["query","args"],"msg":"Field required","input":null},{"type":"missing","loc":["query","kwargs"],"msg":"Field required","input":null}]}'

**Comment by dnutiu** [comment]
So basically parameters defined in dependency overrides get picked up as query parameters:

```python
def auth_service_mock_dependency(x):
  pass

api_client.app.dependency_overrides[authentication_service] = auth_service_mock_dependency
```

**Reply by YuriiMotov** [reply, marked as answer]
What is `x`?
You have to specify the type of this parameter and where FastAPI should look for it (Query, Path, Body, etc.)

**Reply by YuriiMotov** [reply]
Please, in order to get help, provide the signatures of the endpoint function you are trying to test and its dependencies (including dependency overrides)

**Reply by dnutiu** [reply]
The dependency override has the following signature:

```python
async def authentication_service(app_settings: Annotated[Settings, Depends(settings)]):
```

And the function I'm trying to test:

```python
@router.get(...)
async def get_rule(
    domain: Annotated[
        str, Query(..., description="The domain associated with the rule.", min_length=1)
    ],
    auth_service: Annotated[AuthenticationService, Depends(authentication_service)],
):
```

I see now why that happens, since `x` is just a placeholder for the settings dependency in the mocked authentication service dependency it gets picked up as a query parameter by FastAPI when it gets overridden by the test:

```python
def auth_service_mock_dependency(x):
  pass

api_client.app.dependency_overrides[authentication_service] = auth_service_mock_dependency
```
